blockio/src-linux/System/FS/BlockIO/Internal.hs

```diff
@@ -53,7 +52,7 @@ ioHasBlockIO hfs  params =
 
 hSetNoCache :: Handle HandleIO -> Bool -> IO ()
 hSetNoCache h b =
-  FS.withOpenHandle "hSetNoCache" (handleRaw h) (flip Unix.writeFcntlNoCache b)
+  FS.withOpenHandle "hSetNoCache" (handleRaw h) (flip Fcntl.fileSetCaching (not b))
```

Comment:
Note that we're setting to `not b`. The old `writeFcntlNoCache` setting the opposite of what the new `fileSetCaching` does